_playwright-tests/UI/TemplateCRUD.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories, navigateToTemplates } from './helpers/navHelpers';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+import { deleteAllTemplates } from './helpers/deleteTemplates';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+
+const templateNamePrefix = 'template_CRUD';
+const repoNamePrefix = 'custom_repo-template';
+
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+export const rank = () => Math.floor(Math.random() * 10 + 1).toString();
+export const randomNum = () =>
+  Math.floor(Math.random() * 10 + 1)
+    .toString()
+    .padStart(2, '0');
+
+export const url = `https://stephenw.fedorapeople.org/multirepos/${rank()}/repo${randomNum()}/`;
+
+export const templateName = `${templateNamePrefix}-${randomName()}`;
+
+const smallRHRepo = 'Red Hat CodeReady Linux Builder for RHEL 9 ARM 64 (RPMs)';
+
+test.describe('Templates CRUD', () => {
+  test('Add, Read, update, delete a template', async ({ page }) => {
+    await test.step('Delete any templates and template test repos that exist', async () => {
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+      await deleteAllTemplates(page, `&search=${templateNamePrefix}`);
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      // HMS-5268 There are two buttons on the ZeroState page
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('URL').fill(url);
+      await page.getByRole('button', { name: 'Save', exact: true }).click();
+      const row = await getRowByNameOrUrl(page, `${repoName}`);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+    await test.step('Navigate to templates, ensure the Add content template button can be clicked', async () => {
+      await navigateToTemplates(page);
+      await closePopupsIfExist(page);
```

Comment:
> closePopupsIfExist(page)
You already defined these on line 30 and it needs to be defined only once per `test` not per `test.step`